src/Components/Components/src/EventCallbackFactoryBinderExtensions.cs

```diff
@@ -246,28 +260,83 @@ private static bool ConvertToNullableDecimalCore(object obj, out decimal? value)
             return true;
         }
 
-        private static class EnumConverter<T> where T : struct, Enum
+        private static BindConverter<DateTime> ConvertToDateTime = ConvertToDateTimeCore;
+        private static BindConverter<DateTime?> ConvertToNullableDateTime = ConvertToNullableDateTimeCore;
```

Comment:
Added nullable datetime support (without format strings). We have a separate issue tracking adding format strings broadly.